### PR TITLE
Soft 50 reset password & profile Logic & setup global store

### PR DESCRIPTION
**Pull request by JustYousef2024**

### CODE REVIEW

File: components/InputField/InputField.tsx
Comment (MuhammadSawalhy) [inline components/InputField/InputField.tsx:1]: Check: [SOFT-122 ](https://linear.app/iksirs/issue/SOFT-122/refactor-inputfield-component)

Will be scheduled later! إن شاء الله

File: components/Navbar/ProfileDropdown.tsx [hunk 1]
```diff
@@ -26,19 +29,19 @@ export default function ProfileDropDown() {
...
             className="flex items-center gap-x-3.5 py-2 px-3 rounded-lg text-sm text-gray-800 hover:bg-gray-100 focus:ring-2 focus:ring-blue-500 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-gray-300"
-            href="#"
+            href="/login"
           >
             <LuLogOut />
-            Logout
-          </a>
+            <button onClick={logout}>Logout</button>
+          </Link>
         </div>
```
Comment (MuhammadSawalhy) [inline components/Navbar/ProfileDropdown.tsx:45]: How is this possible? `Link` and `button` both work together after a single click! Think about race conditions which is a common problem in software engineering in many fields and technologies.

**Review by MuhammadSawalhy — CHANGES_REQUESTED** [review]
More suggested changes:

Rename these files:

- `ConfirmPIn.tsx` -> `ConfirmPin.tsx`
- `RegisterALLThreeStep.tsx` -> `RegisterAllThreeStep.tsx`

File: components/Gender/DisplayGender.tsx [hunk 2]
```diff
@@ -1,68 +1,43 @@
...
   return (
     <div className="flex justify-between mt-5">
       <p className="w-2/12">Gender</p>
 
       <ul className="flex flex-col sm:flex-row w-9/12">
-        <li className="border-2 w-1/3 inline-flex items-center gap-x-2.5 py-3 px-4 text-sm font-medium border-gray-200 shadow-sm  focus:z-10 focus:border-gray-300 focus:ring-gray-300  -mt-px first:rounded-t-lg first:mt-0 last:rounded-b-lg sm:-ms-px sm:mt-0 sm:first:rounded-se-none sm:first:rounded-es-lg sm:last:rounded-es-none sm:last:rounded-se-lg dark:bg-zinc-900 dark:border-gray-700 dark:text-gray-400 dark:focus:ring-gray-600">
+      {genders.map((gender,indx ) => (
+        
+        <li key={`${indx} - ${gender}`}
+        className="border-2 w-1/3 inline-flex items-center gap-x-2.5 py-3 px-4 text-sm font-medium border-gray-200 shadow-sm  focus:z-10 focus:border-gray-300 focus:ring-gray-300  -mt-px first:rounded-t-lg first:mt-0 last:rounded-b-lg sm:-ms-px sm:mt-0 sm:first:rounded-se-none sm:first:rounded-es-lg sm:last:rounded-es-none sm:last:rounded-se-lg dark:bg-zinc-900 dark:border-gray-700 dark:text-gray-400 dark:focus:ring-gray-600">
```
[the remark this answers is not in the document]
Comment (mohamedahmed-cloud) [reply components/Gender/DisplayGender.tsx:18]: I used index and current element not index only

File: customHooks/Profile/useProfile.ts
[the remark this answers is not in the document]
Comment (mohamedahmed-cloud) [reply customHooks/Profile/useProfile.ts:1]: might used outside the profile itself like a user card. it needs the first & last name and user image. If we didn't use it outside the profile i will move it.

File: customHooks/Profile/useProfileTypesAndFunction.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  date_of_birth: string;
+  timezone: string;
+  gender: string;
+  [key: string]: string;
+}
+
+// Functions
+
+// Handle Update Profile data
+const newObject = {} as updateUserProfileType;
```
[the remark this answers is not in the document]
Comment (mohamedahmed-cloud) [reply customHooks/Profile/useProfileTypesAndFunction.ts:18]: why do you hate type assertion 😂

File: .env
[the remark this answers is not in the document]
Comment (mohamedahmed-cloud) [reply .env:1]: I know but I didn't work on my local machine, I work in Codespace,  Is there any solution for this situation

File: app/[locale]/(auth)/register/RegisterThridStep.tsx [hunk 4]
```diff
@@ -22,14 +27,7 @@ interface PinCodeDataType {
...
 }: RegisterConfirmEmailType) {
-  const [pinCode, setPinCode] = useState<PinCodeDataType>({
-    pin1: '',
-    pin2: '',
-    pin3: '',
-    pin4: '',
-    pin5: '',
-    pin6: '',
-  });
+  const [pinCode, setPinCode] = useState<PinCodeDataType>({} as PinCodeDataType);
```
[the remark this answers is not in the document]
Comment (mohamedahmed-cloud) [reply app/[locale]/(auth)/register/RegisterThridStep.tsx:30]: So, What is the importance of typescript if I make most of the type optional? why we didn't use type assertion to force developers to pass necessary data?

File: components/Buttons/Button.tsx [hunk 5]
```diff
@@ -10,25 +13,28 @@ interface buttonProps {
...
 
+
 export default function AuthButton({
   title,
   width,
   height,
   background,
   ButtonClassName,
   type,
+  disable,
```
[the remark this answers is not in the document]
Comment (MuhammadSawalhy) [reply components/Buttons/Button.tsx:27]: Change the name from `disable` to `disabled`!

**Pull request merged**